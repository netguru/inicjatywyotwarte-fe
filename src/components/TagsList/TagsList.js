import React, { useState, useEffect } from 'react'
import styled from 'styled-components'
import Tag from 'components/Tag/Tag'

const TagsContainer = styled.div.attrs(props => ({ 'data-cy': 'tags-container' }))`
  display: flex;
  align-items: center;
  height: 100%;
  flex-wrap: wrap;

  & > * {
    margin-right: 10px;
  }
`

const Tags = ({
  tags,
  initialActiveTags,
  handleActiveTagsUpdate,
  isTagsListOpen
}) => {
  const [activeTags, setActiveTags] = useState(initialActiveTags || [])

  useEffect(() => {
    handleActiveTagsUpdate(activeTags)
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [activeTags])

  useEffect(() => {
    if (!initialActiveTags.length && activeTags.length) {
      setActiveTags([])
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [initialActiveTags])

  const handleTagOnClick = tag => {
    if (!activeTags.includes(tag)) {
      setActiveTags(activeTags.concat([tag]))
    } else {
      setActiveTags(activeTags.filter(activeTag => activeTag !== tag))
    }
  }

  if (!tags) {
    return <span>Brak tagÃ³w</span>
  }

  const sortedTags = [...tags].sort().sort((a, b) => {
    if (activeTags.includes(a) && !activeTags.includes(b)) {
      return -1
    }
    if (!activeTags.includes(a) && activeTags.includes(b)) {
      return 1
    }
    return 0
  })

  return (
    <TagsContainer>
      {sortedTags.map(tag => {
        const isTagActive = activeTags.includes(tag)
        return (
          <Tag
            active={isTagActive}
            handleOnClick={() => handleTagOnClick(tag)}
            key={tag}
            tag={tag}
            isTagsListOpen={isTagsListOpen}
          />
        )
      })}
    </TagsContainer>
  )
}

export default Tags
