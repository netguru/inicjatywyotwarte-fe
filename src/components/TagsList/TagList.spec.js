import React from 'react'
import TagsList from './TagsList'
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('TagsList', () => {
  test('renders', () => {
    let wrapper = shallow(<TagsList />);
    expect(wrapper.text()).toEqual('Brak tagÃ³w');
  });

  // test('renders tags list', () => {
  //   const tags = ['tag1', 'tag2', 'tag3'];
  //   let wrapper = shallow(<ResourceTile tags={tags} />);
  //
  //   expect(wrapper.find('Tag').length).toEqual(3);
  // });

  test('match snapshot', () => {
    let wrapper = shallow(<TagsList />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});




