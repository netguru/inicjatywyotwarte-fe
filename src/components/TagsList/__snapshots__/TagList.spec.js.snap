// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsList match snapshot 1`] = `
<span>
  Brak tag√≥w
</span>
`;
